kong/plugins/rate-limiting/handler.lua

```diff
@@ -74,6 +74,11 @@ local function get_identifier(conf)
     if req_path == conf.path then
       identifier = req_path
     end
+
+  elseif conf.limit_by == "subdomain" then
+    local host = kong.request.get_host()
+    local subdomain = host:match("(.+)%.")
```

Comment:
Do we need a more robust way to extract subdomains? Some local hostnames are without a dot; some local domains are ended with a dot.